detox/runners/jest/DetoxLifecycleAdapter.js

```diff
@@ -23,6 +33,7 @@ class DetoxJestAdapter /* implements JasmineReporter */ {
 
   async afterAll() {
     await this._flush();
+    await this.detox.cleanup();
```

Comment:
I raised my concerns before, that I would not mix two responsibilities here.
Interesting thought experiment — if you remove all lines in `e2e/init.js` related to Detox+Jest adapter, it will still work and even generate some console and device logs. On the other hand, in the suggested implementation, removing the adapter will not deallocate the resources, preventing exiting.

In any case, if we remove `beforeAll` due to the reasons above, then `afterAll` responsibility might start looking asymmetric. Not sure if it is a good idea.